template/FormlessTemplate.purs

```diff
@@ -0,0 +1,148 @@
+  | Receive Input
+
+data Query a
+  = Command a
+  | Reply (Unit -> a)
+
+type Message = FormFields
+type ChildSlots =
+  ()
+type SelfSlot index = F.Slot Form Query ChildSlots Message index
+
+component
+  :: forall m
+   . MonadAff m
+  => F.Component Form Query ChildSlots Input FormFields m
+component = F.component mkInput $ F.defaultSpec
+  { render = render
+  , handleAction = handleAction
+  , handleEvent = handleEvent
+  , handleQuery = handleQuery
+  , initialize = Just Initialize
+  , finalize = Just Finalize
+  , receive = Just <<< Receive
+  }
+  where
+  -- Converts the Input value passed in by the parent component
+  -- into the Formless' Input value.
+  mkInput :: Input -> F.Input Form AddedState m
+  mkInput _ =
+    -- the two values here are for Formless
+    { validators: Form
+        { name: minLength 7
+        }
+    , initialInputs: Nothing -- when Nothing, will use `Initial` type class
+
+    -- everything else below comes from our `AddedState` rows:
+    , additionalState: Just 5
+    }
+
+  render
+    :: F.PublicState Form AddedState
+    -> F.ComponentHTML Form Action ChildSlots m
+  render st =
+    HH.div_
+      [ HH.p_ [ HH.text $ "Validity: " <> show st.validity ]
+      , HH.p_ [ HH.text $ "Dirty: " <> show st.dirty ]
+      , HH.p_ [ HH.text $ "Being Submitted: " <> show st.submitting ]
+      , HH.p_ [ HH.text $ "Number of Errors: " <> show st.errors ]
+      , HH.p_ [ HH.text $ "Number of Submit attempts: " <> show st.submitAttempts ]
+      , HH.p_ [ HH.text $ "Additional state was: " <> show st.additionalState ]
+
+      , HH.div_
+        [ HH.text "Label" ]
+      , HH.input
+        [ HP.type_ InputText
+        , HP.placeholder "Michael"
+
+        -- gets the value of `_name` in the form's state
+        , HP.value (F.getInput _name st.form)
+
+        -- sets the value of `_name` and then validates it
+        , HE.onValueInput (Just <<< F.setValidate _name)
+        ]
+      , HH.button
+        [ if st.submitting || st.validity /= F.Valid
+            then HP.disabled true
+            else HE.onClick \_ -> Just F.submit
+        ]
+        [ HH.text "Submit" ]
+      ]
+
+  -- Decide what, if anything, to do when Formless events occur.
+  -- For example, if you would like to raise events as messages,
+  -- then use `F.raiseResult` as your `handleEvent` function.
+  handleEvent
+    :: F.Event Form AddedState
+    -> F.HalogenM Form AddedState Action ChildSlots Message m Unit
+  handleEvent = case _ of
+    F.Submitted formContent -> do
+      -- This is how to get the output values of the form.
+      let formFields = F.unwrapOutputFields formContent
+
+      -- We won't do this here, but this is how most will handle a form
+      -- submission: raise it as an event to their parent.
+      --      H.raise formFields
+
+      -- Alternatively, one could do something custom with the output values.
+
+      -- This line exists so the code compiles.
+      pure unit
+    F.Changed formState -> do
```

Comment:
Indicates that the form's content has been changed. This event is triggered anytime a field is changed, whether it passes validation or not.